### ISSUE TITLE
Feature request: Rule name on ESLint errors/warnings  

### ISSUE THREAD

**Issue by zlksnk** [williamboman/nvim-lsp-installer#20]
Here for example the rule name is `react-hooks/exhaustive-deps` but it does not appear in warning. 

Neovim + LSP Saga + Nvim LSP Installer:

![image](https://user-images.githubusercontent.com/26769888/119251040-227c7f80-bb59-11eb-853b-eae7358a73bf.png)

It does appears when using VSCode's ESLint plugin or with Sublime Text's LSP ESLint plugin however.

VSCode + ESLint:

![image](https://user-images.githubusercontent.com/26769888/119251160-e0077280-bb59-11eb-8398-60c8a734e5ee.png)

Sublime Text + LSP ESLint:

![image](https://user-images.githubusercontent.com/26769888/119251162-e72e8080-bb59-11eb-9d9a-a31a473aed00.png)

**Comment by zlksnk** [comment]
Not sure if it is applicable but [compact format](https://eslint.org/docs/user-guide/formatters/#compact) does give rule name at the end of the error/warning description. Deemphasizing the rule name like VS Code does (or merging rule doc in the rule itself which Sublime does) needs to be manually handled though.

**Comment by williamboman** [comment]
Hey! This should entirely depend on the UI code that renders the diagnostics. Refer to the `code` field [`Diagnostic`](https://microsoft.github.io/language-server-protocol/specification#diagnostic) interface definition. Maybe LSP saga already supports this as an option?

**Comment by williamboman** [comment]
I have [custom code](https://github.com/williamboman/nvim-config/blob/main/lua/wb/lsp/diagnostics.lua) that renders these diagnostics, but here's an example how it could look like:

 <img width="454" alt="Screenshot 2021-05-24 at 22 49 40" src="https://user-images.githubusercontent.com/6705160/119405949-5c2cc200-bce2-11eb-8de5-806628846315.png">

The diff is

```diff
diff --git a/lua/wb/lsp/diagnostics.lua b/lua/wb/lsp/diagnostics.lua
index b5a6afb..6a08e7a 100644
--- a/lua/wb/lsp/diagnostics.lua
+++ b/lua/wb/lsp/diagnostics.lua
@@ -167,7 +167,7 @@ function M.show_line_diagnostics(bufnr, line_nr, client_id)
     if vim.tbl_isempty(line_diagnostics) then return end
 
     for i, diagnostic in ipairs(line_diagnostics) do
-        local prefix = string.format("%d. (%s) ", i, diagnostic.source or 'unknown')
+        local prefix = string.format("%d. (%s, %s) ", i, diagnostic.source or 'unknown', diagnostic.code or "")
         local hiname = floating_severity_highlight_name[diagnostic.severity]
         assert(hiname, 'unknown severity: ' .. tostring(diagnostic.severity))
 

```

**Comment by zlksnk** [comment]
Oh that's helpful! I'm moving away from LSPSaga and preferring inbuilt LSP calls and Telescope as LSP helper UIs. I'll refer to yours code and reopen the issue if needed. Thanks again!

**Comment by zlksnk** [comment]
Heya! I've noticed that with your custom code (and also with inbuilt NVim UI code) for line diagnostics, each diagnostic line that is shown can get clipped off because it doesn't get wrapped. Upon further investigation I found out that using `:set wrap` before calling line diagnostics can wrap lines in the floating popup but then that's a global setter. 

How can I solve this? I want to keep nowrap as global but wrap lines in floating window.
